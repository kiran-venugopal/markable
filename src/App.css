html,
body,
.App,
#root {
  height: 100%;
}
@import url("https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap");

body {
  margin: 0;
  font-family: "Noto Sans", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  letter-spacing: 1px;
}

* {
  font-family: "Noto Sans", sans-serif;
}

.center-aligned {
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.heading-text {
  font-size: 19px;
  margin: 30px 0;
}
label.checkbox {
  font-size: 14px;
  margin: 10px 0;
  display: flex;
  align-items: center;
}
label.checkbox input {
  margin: 0 8px;
}

.react-grid-item {
  box-shadow: rgb(0 0 0 / 28%) 1px 2px 7px;
  border: none !important;
}

.ql-container {
  height: calc(100% - 42px) !important;
}

.toast-element {
  color: white !important;
  font-weight: 600 !important;
}
