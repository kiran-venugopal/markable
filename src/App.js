import "./App.css";
import { BrowserRouter, Switch } from "react-router-dom";
import HomePage from "./pages/HomePage";
import { useRecoilState } from "recoil";
import { userState } from "./recoil/atoms";
import GuardedRoute from "./utils/RouteGuard";
import loginPage from "./pages/LoginPage";
import {
  ToastsContainer,
  ToastsContainerPosition,
  ToastsStore,
} from "react-toasts";

function App() {
  const [userData] = useRecoilState(userState);

  return (
    <div className="App">
      <BrowserRouter>
        <Switch>
          <GuardedRoute
            onlyUnauthorized={true}
            path="/login"
            component={loginPage}
            exact
          />
          <GuardedRoute
            path="/"
            component={HomePage}
            auth={userData.isLoggedIn}
          />
        </Switch>
      </BrowserRouter>
      <ToastsContainer
        position={ToastsContainerPosition.BOTTOM_CENTER}
        store={ToastsStore}
      />
    </div>
  );
}

export default App;
