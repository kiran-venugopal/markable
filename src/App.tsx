import "./App.css";
import React from "react";
import Navbar from "./components/navbar";
import Sidebar from "./components/sidebar";
import EditorSection from "./components/editor-section";
import { RecoilRoot } from "recoil";

function App() {
  return (
    <RecoilRoot>
      <div className="App">
        <Navbar />
        <div className="editor-container">
          <Sidebar />
          <EditorSection />
        </div>
      </div>
    </RecoilRoot>
  );
}

export default App;
