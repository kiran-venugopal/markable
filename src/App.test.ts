import { cleanup, render, screen } from "@testing-library/react";
import App from "./App";

afterEach(cleanup);

test("should take a snapshot", () => {


  // @ts-expect-error ts-migrate(7022) FIXME: 'asFragment' implicitly has type 'any' because it ... Remove this comment to see the full error message
  const { asFragment } = render(<App />);


  // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
  expect(asFragment(<App />)).toMatchSnapshot();
});

test("renders copy markdown button", () => {


  // @ts-expect-error ts-migrate(2749) FIXME: 'App' refers to a value, but is being used as a ty... Remove this comment to see the full error message
  render(<App />);
  const linkElement = screen.getByText(/copy markdown/i);
  expect(linkElement).toBeInTheDocument();
});
