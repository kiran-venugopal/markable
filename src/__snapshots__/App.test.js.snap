// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar"
    >
      <div
        class="logo"
      >
        <svg
          height="25"
          width="25"
        >
          logo.svg
        </svg>
      </div>
      <div
        class="file-name"
        data-testid="file-name-label"
      >
        untitled.md
      </div>
      <button
        class="primary"
      >
        <div
          class="icon"
        >
          <svg
            height="11"
            width="11"
          >
            copy.svg
          </svg>
        </div>
        <div
          class="label"
        >
          Copy Markdown
        </div>
      </button>
    </div>
    <div
      class="editor-container"
    >
      <div
        class="sidebar"
        style="width: 270px;"
      >
        <div
          class="sidebar-actions"
        >
          <button
            class="icon-button github"
          >
            <svg>
              github.svg
            </svg>
          </button>
          <div
            style="display: flex;"
          >
            <button
              class="icon-button new-folder"
              title="new folder"
            >
              <svg
                height="17"
                width="17"
              >
                new-folder.svg
              </svg>
            </button>
            <button
              class="icon-button new-file"
              title="new file"
            >
              <svg
                height="17"
                width="17"
              >
                new.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="folders"
        >
          <div
            class="dir-item active"
          >
            <div
              class="icon"
            >
              <svg
                height="20"
                width="20"
              >
                md.svg
              </svg>
            </div>
            <div
              class="name"
            >
              untitled.md
            </div>
            <div
              class="actions"
            >
              <button
                class="icon-button"
              >
                <svg
                  height="18"
                  width="18"
                >
                  remove.svg
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="auth-section"
        >
          <div
            style="overflow: hidden;"
          >
            <button
              class="secondary"
            >
              Sign In
            </button>
          </div>
        </div>
        <div
          class="resizer"
        />
      </div>
      <div
        class="editor-section"
        spellcheck="false"
      >
        <div
          class="sc-bdvvtL bMzWfw editor"
          dir="auto"
        >
          <div
            class="sc-dJjYzT klaQDU"
            dir="auto"
          >
            <div
              class="ProseMirror"
              contenteditable="true"
              role="textbox"
            >
              <button
                class="block-menu-trigger ProseMirror-widget"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="24px"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13,11 L13,6 C13,5.44771525 12.5522847,5 12,5 C11.4477153,5 11,5.44771525 11,6 L11,6 L11,11 L6,11 C5.44771525,11 5,11.4477153 5,12 C5,12.5522847 5.44771525,13 6,13 L11,13 L11,18 C11,18.5522847 11.4477153,19 12,19 C12.5522847,19 13,18.5522847 13,18 L13,13 L18,13 C18.5522847,13 19,12.5522847 19,12 C19,11.4477153 18.5522847,11 18,11 L13,11 Z"
                  />
                </svg>
              </button>
              <p
                class="placeholder"
                data-empty-text="Write something niceâ€¦"
              >
                <br />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
